contracts/BitGuildMarketplace.sol

```diff
@@ -0,0 +1,334 @@
+    using SafeMath for uint256;
+
+    // Callback values from zepellin ERC721Receiver.sol
+    // Old ver: bytes4(keccak256("onERC721Received(address,uint256,bytes)")) = 0xf0b9e5ba;
+    bytes4 constant ERC721_RECEIVED_OLD = 0xf0b9e5ba;
+    // New ver w/ operator: bytes4(keccak256("onERC721Received(address,address,uint256,bytes)")) = 0xf0b9e5ba;
+    bytes4 constant ERC721_RECEIVED = 0x150b7a02;
+
+    // BitGuild Contracts
+    // BitGuildToken public PLAT = BitGuildToken(0x7E43581b19ab509BCF9397a2eFd1ab10233f27dE); // Main Net
+    // BitGuildWhitelist public Whitelist = BitGuildWhitelist(); // Main Net
+    // BitGuildFeeProvide public FeeProvider = BitGuildFeeProvider(); // Main Net
+    BitGuildToken public PLAT = BitGuildToken(0x0F2698b7605fE937933538387b3d6Fec9211477d); // Rinkeby
+    BitGuildWhitelist public Whitelist = BitGuildWhitelist(0x72b93A4943eF4f658648e27D64e9e3B8cDF520a6); // Rinkeby
+    BitGuildFeeProvider public FeeProvider = BitGuildFeeProvider(0x9bE88C776299795A4996D351215274F3f1d84100); // Rinkeby
+
+    uint public defaultExpiry = 7 days;             // default expiry is 7 days
+    bool public allowWithdrawBeforeExpiry = false;  // allow withdraw listing before expiry? default is false
+
+    enum Currencies { PLAT, ETH }
+    struct Listing {
+        Currencies currency;    // ETH or PLAT
+        address seller;         // seller address
+        address token;          // token contract
+        uint tokenId;           // token id
+        uint price;             // Big number in ETH or PLAT
+        uint createdAt;         // timestamp
+        uint expiry;            // createdAt + defaultExpiry
+    }
+
+    mapping(bytes32 => Listing) public listings;
+
+    event LogListingCreated(address _seller, address _contract, uint _tokenId, uint _createdAt, uint _expiry);
+    event LogListingExtended(address _seller, address _contract, uint _tokenId, uint _createdAt, uint _expiry);
+    event LogItemSold(address _buyer, address _seller, address _contract, uint _tokenId, uint _price, Currencies _currency, uint _soldAt);
+    event LogItemWithdrawn(address _seller, address _contract, uint _tokenId, uint _withdrawnAt);
+    event LogItemExtended(address _contract, uint _tokenId, uint _modifiedAt, uint _expiry);
+
+    modifier onlyWhitelisted(address _contract) {
+        require(Whitelist.isWhitelisted(_contract), "Contract not in whitelist.");
+        _;
+    }
+
+    // @dev fall back function
+    function () external payable {
+        revert();
+    }
+
+    // @dev Retrieve hashkey to view listing
+    function getHashKey(address _contract, uint _tokenId) public pure returns(bytes32 key) {
+        key = _getHashKey(_contract, _tokenId);
+    }
+
+    // ===========================================
+    // Fee functions (from fee provider contract)
+    // ===========================================
+    // @dev get default fee
+    function getFee(uint _price) public view returns(uint percent, uint fee) {
```

Comment:
Same here, I don't think you want the default like at all - that could lead to misuse